src/FluentAssertions.Analyzers/Utilities/FluentAssertionsAnalyzer.cs

```diff
@@ -62,9 +62,11 @@ protected virtual Diagnostic AnalyzeExpression(ExpressionSyntax expression, Sema
             var variableNameExtractor = new VariableNameExtractor(semanticModel);
             expression.Accept(variableNameExtractor);
 
-            if (variableNameExtractor.VariableIdentifierName == null) return null;
-            var typeInfo = semanticModel.GetTypeInfo(variableNameExtractor.VariableIdentifierName);
-            if (!ShouldAnalyzeVariableTypeCore(typeInfo.Type, semanticModel)) return null;
+            if (variableNameExtractor.PropertiesAccessed
```

Comment:
need to add more tests for edge cases where some nested properties are not really interesting (specifically `Count` and a few more)